sdk/src/signature.rs

```diff
@@ -160,6 +161,57 @@ impl KeypairUtil for Keypair {
     }
 }
 
+impl<T> PartialEq<T> for Keypair
+    where T: KeypairUtil
+{
+    fn eq(&self, other: &T) -> bool {
+        self.pubkey() == other.pubkey()
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct Presigner {
+    pubkey: Pubkey,
+    signature: Signature,
+}
+
+impl Presigner {
+    fn new(pubkey: &Pubkey, signature: &Signature) -> Self {
+        Self {
+            pubkey: *pubkey,
+            signature: *signature,
+        }
+    }
+}
+
+#[derive(Debug, Error, PartialEq)]
+enum PresignerError {
+    #[error("pre-generated signature cannot verify data")]
+    VerificationFailure,
+}
+
+impl KeypairUtil for Presigner {
+    fn try_pubkey(&self) -> Result<Pubkey, Box<dyn error::Error>> {
+        Ok(self.pubkey)
+    }
+
+    fn try_sign_message(&self, message: &[u8]) -> Result<Signature, Box<dyn error::Error>> {
+        if self.signature.verify(self.pubkey.as_ref(), message) {
+            Ok(self.signature)
+        } else {
+            Err(PresignerError::VerificationFailure.into())
+        }
+    }
+}
+
+impl<T> PartialEq<T> for Presigner
+    where T: KeypairUtil
+{
+    fn eq(&self, other: &T) -> bool {
+        self.pubkey() == other.pubkey()
```

Comment:
This doesn't seem quite right to me. Shouldn't the signature fields be equivalent as well?

Reply:
I based this off the `KeypairEq` wrapper in CLI.  I don't think we'll be able to arbitrarily compare the sigs of a `KeypairUtil` since most of the implementer will need data passed to generate one

Reply:
Oh, I see, I missed the generic. Do we need PartialEq between all KeypairUtil types?
The current implementation is just between one Keypair and another.

Reply:
Are you thinking that would be helpful when compiling/checking transactions with mixed signer types?

Reply:
Yes.  That and there's currently a `PartialEq` dependency on anything that is held in a `CliCommand` value field

Reply:
I reserve the right to change my mind on this later, but okay for now, then.

Reply:
I think we can dump it if/when we convert `CliCommand` to a `trait`